tests/trainer/test_trainer.py

```diff
@@ -110,17 +110,21 @@ class TestTrainerEquivalence():
     reference_folder: pathlib.Path
 
     def assert_models_equal(self, model_1, model_2):
+        assert id(model_1) != id(model_2), "Same model should not be compared."
         for param1, param2 in zip(model_1.parameters(), model_2.parameters()):
             torch.testing.assert_allclose(param1, param2)
 
-    @pytest.fixture
+    @pytest.fixture(scope="function")
     def config(self, device, precision, world_size):
+        """ Returns the reference config. """
+
+        seed_all(seed=0)
         return {
             'model': SimpleModel(),
             'train_dataloader': DataLoader(
                 dataset=RandomClassificationDataset(),
                 batch_size=4,
-                shuffle=True,
+                shuffle=False,
```

Comment:
not that it makes a big difference, but if we're seeding, then shuffle should still be able to be True?

Reply:
Actually, having some trouble getting this to pass with `True`. More specifically, if I do:
```
seed_all(0)
dl1 = DataLoader(RandomClassificationDataset(), shuffle=True)
seed_all(0)
dl2 = DataLoader(RandomClassificationDataset(), shuffle=True)

torch.testing.assert_allclose(next(dl1.__iter__()), next(dl2.__iter())
```
fails for `shuffle=True`, but passes for `shuffle=False`. Is there a worker-specific rng that needs to be set?

Reply:
Ahhh...the dataloader is seeded on the first spin, not when it is initialized.